src/graph/GraphViewsManager.ts

```diff
@@ -3,45 +3,80 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import * as fs from "fs";
 import * as path from "path";
 import * as vscode from 'vscode';
-import { resourcesPath } from "../constants";
 import { areConfigsEqual, GraphConfiguration } from './GraphConfiguration';
 import { GraphViewServer } from './GraphViewServer';
 
-const scheme = "vscode-cosmosdb-graphresults";
-const previewBaseUri = scheme + '://results/';
+// const scheme = "vscode-cosmosdb-graphresults";
+// const previewBaseUri = scheme + '://results/';
 
 interface IServerProvider {
   findServerById(id: number): GraphViewServer;
 }
 
-export class GraphViewsManager implements IServerProvider {
+interface IGraphClientHtmlPaths {
+  d3Uri: string;
+  socketUri: string;
+  graphClientUri: string;
+  cosmosIconUri: string;
+  cssUri: string;
+}
+
+export class GraphViewsManager implements IServerProvider { //Graphviews Panel
   private _lastServerId = 0;
 
   // One server (and one HTML view) per graph, as represented by unique configurations
   private _servers = new Map<number, GraphViewServer>(); // map of id -> map
+  private _panels = new Map<number, vscode.WebviewPanel>(); // map of id -> map
+  private _panelViewType: string = "GraphExplorer";
+
+  constructor(private _context: vscode.ExtensionContext) {
 
-  public constructor(private _context: vscode.ExtensionContext) {
-    let documentProvider = new GraphViewDocumentContentProvider(this);
-    let registration = vscode.workspace.registerTextDocumentContentProvider(scheme, documentProvider);
-    this._context.subscriptions.push(registration);
   }
 
   public async showGraphViewer(
     tabTitle: string,
     config: GraphConfiguration
   ): Promise<void> {
-    try {
-      let id = await this.getOrCreateServer(config);
-
-      // Add server ID to the URL so that GraphViewDocumentContentProvider knows which port to use in the HTML
-      let serverUri = previewBaseUri + id.toString();
-      await vscode.commands.executeCommand('vscode.previewHtml', vscode.Uri.parse(serverUri), vscode.ViewColumn.One, tabTitle);
-    } catch (error) {
-      vscode.window.showErrorMessage(error.message || error);
+    let id = await this.getOrCreateServer(config);
+
+    let existingPanel: vscode.WebviewPanel;
+    this._panels.forEach((p, key) => {
```

Comment:
You can use this._panels.find()

Reply:
I don't think Maps have a `find` function, and that's why getOrCreateServer too does a `forEach`.

Reply:
Map.has()/get().  Wouldn't be very useful as a map without that.

getOrCreateServer needs to find via areConfigsEqual, not key

Reply:
Isn't this just findPanelById?